url78f6619d18_IntersectAustralia_rifcs_tgz-pJ8-org_ands_rifcs_example_SampleRIFCSBuildCollectionJ8/src/SampleRIFCSRead.java

```diff
@@ -0,0 +1,57 @@
+/*    */ package org.ands.rifcs.example;
+/*    */ 
+/*    */ import java.io.FileInputStream;
+/*    */ import java.io.FileNotFoundException;
+/*    */ import java.io.IOException;
+/*    */ import java.io.PrintStream;
+/*    */ import java.net.MalformedURLException;
+/*    */ import java.util.Iterator;
+/*    */ import java.util.List;
+/*    */ import javax.xml.parsers.ParserConfigurationException;
+/*    */ import org.ands.rifcs.base.Collection;
+/*    */ import org.ands.rifcs.base.Name;
+/*    */ import org.ands.rifcs.base.NamePart;
+/*    */ import org.ands.rifcs.base.RIFCS;
+/*    */ import org.ands.rifcs.base.RIFCSException;
+/*    */ import org.ands.rifcs.base.RIFCSWrapper;
+/*    */ import org.ands.rifcs.base.RegistryObject;
+/*    */ import org.ands.rifcs.ch.RIFCSReader;
+/*    */ import org.xml.sax.SAXException;
+/*    */ 
+/*    */ public class SampleRIFCSRead
+/*    */ {
+/* 39 */   private static RIFCS rifcs = null;
+/*    */ 
+/*    */   public static void main(String[] args) throws RIFCSException, FileNotFoundException, SAXException, ParserConfigurationException, IOException, MalformedURLException
+/*    */   {
+/* 43 */     RIFCSReader rr = new RIFCSReader();
+/* 44 */     rr.mapToDOM(new FileInputStream(args[0]));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor: *FileInputStream*. Currently, there are execution paths that do not contain closure statements, for example, when RIFCSReader.mapToDOM throws an exception. Ensure the object returned by *FileInputStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).